src/app.js

```diff
@@ -18,15 +18,39 @@ import Routes from './Routes';
 import config from './config';
 
 // If you want to change the language, Change the imports to the match
-// the wanted locale.
+// the wanted locale. Use the messagesInLocale import to add the translations.
 //
 // Remember to also change the language in the config.js file.
 import localeData from 'react-intl/locale-data/en';
-import messages from './translations/en.json';
+import messagesInLocale from './translations/en.json';
+
 // If you are using a non-english locale with moment library,
 // you should also import time specific formatting rules for that locale
 // e.g. for French: import 'moment/locale/fr';
 
+// Flex template application uses English translations as default.
+// If translation key is missing from e.g. fr.json corresponding key
+// will be added to messages from en.json to prevent missing translation key errors.
+import defaultMessages from './translations/en.json';
```

Comment:
Now that I read these comments, I think that these lines should be reordered a bit:

```js
// Flex template application uses English translations as default.
import defaultMessages from './translations/en.json';

// If you want to change the language, change the imports to match the wanted locale:
//   1) Change the language in the config.js file!
//   2) Import correct locale rules for React Intl library
//   3) Import locale rules for Moment library
//   4) Use the `messagesInLocale` import to add the translation file.

// Step 2:
// Import locale rules for React Intl library 
import localeData from 'react-intl/locale-data/en';

// Step 3:
// If you are using a non-english locale with moment library,
// you should also import time specific formatting rules for that locale
// e.g. for French: import 'moment/locale/fr';

// Step 4:
// If you are using a non-english locale, point `messagesInLocale` to correct .json file
import messagesInLocale from './translations/en.json';

// If translation key is missing from `messagesInLocale` (e.g. fr.json), 
// corresponding key will be added to messages from `defaultMessages` (en.json) 
// to prevent missing translation key errors.
```

Reply:
Also docs should reflect this step-by-step model ^